apps/web/src/components/templates/in-app-editor/ContentContainer.tsx

```diff
@@ -17,30 +18,11 @@ export function ContentContainer({
 }) {
   const theme = useMantineTheme();
   const ref = useRef<HTMLDivElement>(null);
-  const [visiblePlaceholder, setVisiblePlaceholder] = useState(!!value);
+  const showPlaceHolder = value.length === 0;
   const [content, setContent] = useState<string | IEmailBlock[]>(value);
 
-  const handleChange = (e: React.ChangeEvent<HTMLDivElement>) => {
-    const textContent = e.target.textContent;
-    if (ref?.current) {
-      ref.current.textContent = textContent;
-    }
-
-    let showPlaceHolder = !textContent;
-    if (textContent === '<br>') showPlaceHolder = true;
-    setVisiblePlaceholder(showPlaceHolder);
-
-    onChange(textContent);
-  };
-
-  useEffect(() => {
-    let showPlaceHolder = value.length === 0;
-    if (value === '<br>') showPlaceHolder = true;
-    setVisiblePlaceholder(showPlaceHolder);
-  }, [value, content]);
-
   useEffect(() => {
-    if (value !== ref.current?.textContent) {
+    if (ref.current && value !== ref.current?.innerHTML) {
```

Comment:
this effect won't trigger after the `onChange` event, because the `value` and `innerHTML` will be the same